Support parsing mixed %Z and %z

yoh2 opened the pull request:
This PR add a feature parsing mixed `%Z` and `%z` (e.g. `Thu, 20 Oct 2022 02:14:22 -0600 (CST)`) into correct zone information.
That format can be found in the `Date` header of some emails, for example.

itchyny commented:
We cannot use `loc` to extract the `name` and `offset` for parsing the both, because it breaks the behavior of parsing one of them. For example,`timefmt.Parse("2020-07-24 09:07:29 +0900", "%Y-%m-%d %H:%M:%S %z")` should return the time with zone `+0900` with empty name, not `UTC`. Also, `timefmt.ParseInLocation("2020-07-24 09:07:29 UTC", "%Y-%m-%d %H:%M:%S %Z", time.Local)` should return the time with zone `+0000 UTC`, note that the offset is not copied from `time.Local`.

itchyny commented on parse_test.go:
```diff
@@ -743,6 +756,16 @@ func TestParse(t *testing.T) {
 				if !got.Equal(tc.t) {
 					t.Errorf("expected: %v, got: %v", tc.t, got)
 				}
+				if tc.sameZone {
```
We'd better always check the zone equality. Only one existing test case (`2020-07-24 23:14:15 +0000`) should be fixed and then all cases should pass.

yoh2 commented:
I agreed how it should behave when it appears only one of `%Z` or `%z`.
Then, is the behavior below acceptable?

- only `%Z` in the format: `loc`has parsed name and offset `+0000`
- only `%z` in the format: `loc` has empty name and parsed offset
- both `%Z` and `%z` in the format: `loc` has parsed name and parsed offset

--
Even if the behavior is acceptable, should we close this PR and open new issue since the new code should be very different from current code?

itchyny commented:
That behavior looks ok, and you can fix in this branch, but squash the commits anyway.

yoh2 commented:
Sorry for the late reply.

I fixed  the sources as follows:

- fixed parsing behavior as [comment above](https://github.com/itchyny/timefmt-go/pull/4#issuecomment-1304547157)
- added zone comparisons to the parse test
- reconstructed commits

itchyny commented:
Thank you.
